espnet2/gan_tts/vits/wavenet.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2021 Tomoki Hayashi
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""WaveNet modules.
+
+This code is modified from https://github.com/kan-bayashi/ParallelWaveGAN.
+
+"""
+
+import logging
+import math
+
+from typing import Optional
+
+import torch
+
+from espnet2.gan_tts.vits.residual_block import Conv1d1x1
+from espnet2.gan_tts.vits.residual_block import WaveNetResidualBlock as ResidualBlock
+
+
+class WaveNet(torch.nn.Module):
+    """WaveNet with global conditioning."""
+
+    def __init__(
+        self,
+        in_channels: int = 1,
+        out_channels: int = 1,
+        kernel_size: int = 3,
+        layers: int = 30,
+        stacks: int = 3,
+        base_dilation: int = 2,
+        residual_channels: int = 64,
+        aux_channels: int = -1,
+        gate_channels: int = 128,
+        skip_channels: int = 64,
+        global_channels: int = -1,
+        dropout_rate: float = 0.0,
+        bias: bool = True,
+        use_weight_norm: bool = True,
+        use_first_conv: bool = False,
+        use_last_conv: bool = False,
+        scale_residual: bool = False,
+        scale_skip_connect: bool = False,
```

Comment:
@r9y9 How do you think about default value of these parameters?

Reply:
This is reasonable to me. IIRC, there weren't significant differences in my past experiments.